src/main/java/seedu/address/model/person/StudentPredicateList.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.model.person;
+
+import java.util.ArrayList;
+import java.util.function.Predicate;
+
+/**
+ * Represents a list of student predicates.
+ */
+public class StudentPredicateList {
```

Comment:
I like how you create those predicate classes and use StudentPredicateList object as a combination of predicates to filter students as intended by the user.